### ISSUE TITLE
do not compile without aes-ni

### ISSUE THREAD

**Issue by renheren** [enerc/VulkanXMRMiner#2]
>[&nbsp;40%]&nbsp;Building&nbsp;CXX&nbsp;object&nbsp;CMakeFiles/miner.dir/src/slow-hash.cpp.o
In&nbsp;file&nbsp;included&nbsp;from&nbsp;/home/miner/mining/docker/vulkanxmrminer/VulkanXMRMiner/src/slow-hash.cpp:32:0:
/usr/lib/gcc/x86_64-linux-gnu/5/include/wmmintrin.h:&nbsp;In&nbsp;function&nbsp;‘void&nbsp;aes_256_assist2(__m128i*,&nbsp;__m128i*)’:
/usr/lib/gcc/x86_64-linux-gnu/5/include/wmmintrin.h:87:1:&nbsp;error:&nbsp;inlining&nbsp;failed&nbsp;in&nbsp;call&nbsp;to&nbsp;always_inline&nbsp;‘__m128i&nbsp;_mm_aeskeygenassist_si128(__m128i,&nbsp;int)’:&nbsp;target&nbsp;specific&nbsp;option&nbsp;mismatch
&nbsp;_mm_aeskeygenassist_si128&nbsp;(__m128i&nbsp;__X,&nbsp;const&nbsp;int&nbsp;__C)
&nbsp;^
/home/miner/mining/docker/vulkanxmrminer/VulkanXMRMiner/src/slow-hash.cpp:903:43:&nbsp;error:&nbsp;called&nbsp;from&nbsp;here
&nbsp;&nbsp;t4&nbsp;=&nbsp;_mm_aeskeygenassist_si128(*t1,&nbsp;0x00);
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;^
CMakeFiles/miner.dir/build.make:134:&nbsp;recipe&nbsp;for&nbsp;target&nbsp;'CMakeFiles/miner.dir/src/slow-hash.cpp.o'&nbsp;failed
make[2]:&nbsp;***&nbsp;[CMakeFiles/miner.dir/src/slow-hash.cpp.o]&nbsp;Error&nbsp;1
CMakeFiles/Makefile2:104:&nbsp;recipe&nbsp;for&nbsp;target&nbsp;'CMakeFiles/miner.dir/all'&nbsp;failed
make[1]:&nbsp;***&nbsp;[CMakeFiles/miner.dir/all]&nbsp;Error&nbsp;2
Makefile:83:&nbsp;recipe&nbsp;for&nbsp;target&nbsp;'all'&nbsp;failed
make:&nbsp;***&nbsp;[all]&nbsp;Error&nbsp;2

According to https://github.com/Snipa22/xmr-node-proxy/issues/18
It seems it is related to my j1800 do not have aes-ni. is there a workaround for this?

**Comment by enerc** [comment]
I have updated the repository with software AES. Build with:
cmake -DSOFT_AES=1 ..
Please report if this works for you (I don't have non aes CPUs, so something might break elsewhere).

**Comment by renheren** [comment]
Thansk for the fast response!

> Checking your cards
> Index:0 MemorySize:61440 Gb - GeForce GTX 1050
> Index:1 MemorySize:61440 Gb - GeForce GTX 1050 Ti
> 
> Finding best configuration....
> Illegal instruction (core dumped)

It's strange that it didn't detect card memory correctly and finally crashed for "illegal instruction".

**Comment by enerc** [comment]
That was caused by Intel® Celeron® Processor G1610 not having avx support. I have updated the build, and also fixed memory detection on Nvidia as reported on issue #3

**Comment by renheren** [comment]
Thanks! Finally I passed the configuration stage. Now I'm stuck with the same issue with https://github.com/enerc/VulkanXMRMiner/issues/3
Looking forward to your progress!

**Comment by enerc** [comment]
Did you sync with the latest changes (commit memoryOffset must be x alignment (Vulkan Specs))?

**Comment by renheren** [comment]
with commit 50cf40098e7b2d52e93ed1425c518eae46c96f7a, It passed configuration stage successfully!
But finally came to a "Segmentation fault".

```
 "debug_network" : "false",
 "console_listen_port" : "0",
 "console_refresh_rate" : "30",
 "cards" : [
  {
    "index"    : "0",
    "cu"       : "1",
    "factor"   : "1",
    "worksize" : "16"
  },
  {
    "index"    : "1",
    "cu"       : "1",
    "factor"   : "1",
    "worksize" : "16"
  },
  {
    "index"    : "2",
    "cu"       : "1",
    "factor"   : "1",
    "worksize" : "16"
  }
 ]
}

Play with the parameters for optimum hashrate.
 _____        _               _
| ____|_ __  (_) ___  _   _  | |
|  _| | '_ \ | |/ _ \| | | | | |
| |___| | | || | (_) | |_| | |_|
|_____|_| |_|/ |\___/ \__, | (_)
           |__/       |___/

Connecting to pool.supportxmr.com:7777 ... done!
Using XMR Variant: cn/r
Mining started.... (Press q to stop)
[Time] 'Total H/s' 'Good shares'/'Invalid shares'/'Expired shares' [GPU] H/s 'good hashes'/'bad hashes'
[10:59:43] 11705.0H/s 0/0/0 [0]:3819.6H/s 0/0  [1]:3119.0H/s 0/0  [2]:4766.4H/s 0/0
Segmentation fault (core dumped)
```

**Comment by enerc** [comment]
I think there is something wrong with the Vulkan driver.  Can you run vulkaninfo and check the memory reported by this tool?. if it's incorrect try to update to latest Nvidia driver.
cu and factor being 1 also indicates the Vulkan APIs are returning crazy values.

**Comment by renheren** [comment]
Cu and factor being 1 is choosen manually by me. As the return values are indeed crazy big, I do not dare to use those values:
```
Card:0  8 Compute Units/Stream Multiprocessors, using factor 8332016 (66656128 threads), local size 16
Card:1  11 Compute Units/Stream Multiprocessors, using factor 6059648 (66656128 threads), local size 16
Card:2  12 Compute Units/Stream Multiprocessors, using factor 5554672 (66656064 threads), local size 16
```
It's also strange that while I used one 1050 ti and two 1050, none of cu and factor are identical among the three cards.
For the vulkaninfo, I assume you are referring to the following section:
```
        memoryHeaps[0] :
                size          = 4294967296 (0x100000000) (4.00 GiB)
                flags:
                        VK_MEMORY_HEAP_DEVICE_LOCAL_BIT
        memoryHeaps[1] :
                size          = 3020550144 (0xb409f000) (2.81 GiB)
                flags:
                        None

        memoryHeaps[0] :
                size          = 2147483648 (0x80000000) (2.00 GiB)
                flags:
                        VK_MEMORY_HEAP_DEVICE_LOCAL_BIT
        memoryHeaps[1] :
                size          = 3020550144 (0xb409f000) (2.81 GiB)
                flags:
                        None

        memoryHeaps[0] :
                size          = 2147483648 (0x80000000) (2.00 GiB)
                flags:
                        VK_MEMORY_HEAP_DEVICE_LOCAL_BIT
        memoryHeaps[1] :
                size          = 3020550144 (0xb409f000) (2.81 GiB)
                flags:
                        None
```
It seems the memory are correctly detected. BTW, I'm using Nvidia driver 418.43, which is the latest long support branch.

**Comment by enerc** [comment]
Nvidia fixes provided in 0.4.1 release.
